src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+
+```js
+return (
+  <div className={classes}>
+    <LineItemUnitPrice transaction={transaction} unitType={unitType} intl={intl} />
+    <LineItemBookingPeriod transaction={transaction} booking={booking} unitType={unitType} />
+    <LineItemUnitsMaybe transaction={transaction} unitType={unitType} />
+    <LineItemCleaningFeeMaybe transaction={transaction} intl={intl} />
+    ...
+```
+
+While we're working on the booking breakdown, let's also modify the sub
+total calculations so that the cleaning fee is taken into consideration.
+Sub total is shown to a user in the booking breakdown if the transaction
+process is configured to charge commission from the mentioned user's
+role in a given transaction. So if a provider commission is configured
+then providers will see a sub total line in the booking breakdown.
+
+To take the cleaning fee into account when calculatin the sub total, in
+the`LineItemSubTotalMaybe.js` file, import `LINE_ITEM_CLEANING_FEE` from
+`util/types.js` and add the following lines to the bottom of the
+imports:
+
+```js
+import { types as sdkTypes } from '../../util/sdkLoader';
+import Decimal from 'decimal.js';
+import {
+  unitDivisor,
+  convertMoneyToNumber,
+  convertUnitToSubUnit,
+} from '../../util/currency';
+const { Money } = sdkTypes;
+```
+
+Modify the component so that `formattedSubTotal` is resolved as follows:
+
+```js
+const unitPurchase = transaction.attributes.lineItems.find(
+  item => item.code === unitType && !item.reversal
+);
+
+if (!unitPurchase) {
+  throw new Error(
+    `LineItemSubTotalMaybe: lineItem (${unitType}) missing`
+  );
+}
+
+const cleaningFeePurchase = transaction.attributes.lineItems.find(
+  item => item.code === LINE_ITEM_CLEANING_FEE && !item.reversal
+);
+
+const unitNumeric = convertMoneyToNumber(unitPurchase.lineTotal);
+const cleaningFeeNumeric = cleaningFeePurchase
+  ? convertMoneyToNumber(cleaningFeePurchase.lineTotal)
+  : null;
+const numericTotalPrice = cleaningFeeNumeric
+  ? new Decimal(unitNumeric).plus(cleaningFeeNumeric).toNumber()
+  : new Decimal(unitNumeric).toNumber();
+const currency = unitPurchase.lineTotal.currency;
+const subTotal = new Money(
+  convertUnitToSubUnit(numericTotalPrice, unitDivisor(currency)),
+  currency
+);
+
+const formattedSubTotal = formatMoney(intl, subTotal);
+```
+
+Now we have all that we need to enable the customer to add a cleaning
+service to their booking. By now the booking section of your listing
+page should be looking like this:
+
+![Booking dates form](./booking-dates-form.png)
+
+## 5. Add the cleaning fee to the transaction
+
+Now it's time to add the cleaning fee line item to the transaction
+initiation request and then we're all set!
+
+Import `LINE_ITEM_CLEANING_FEE` from `util/types.js` to the
+`CheckoutPage.js` file and update the `customPricingParams` method we
+added in step 3. to resolve the cleaning fee line item based on a
+`cleaningFee` value passed in the method params:
+
+```js
+customPricingParams(params) {
```

Comment:
I have mixed feelings about should steps 3 and 5 be combined. That would reduce this wall of text a bit.
Then order would be: change `ListingPage`, change `CheckoutPage`, add validation to `TransactionPage`.
(i.e. the same order in which transaction process actually goes.)

Reply:
Good idea, now the order is:
- change the current booking to use custom pricing
- update UI for adding the cleaning fee
- add the fee to the tx initiate request

I'd stick to the current order for now. There it's clear that the current bookings are also going to use custom pricing in the future even though it's a bit back and forth between the files.